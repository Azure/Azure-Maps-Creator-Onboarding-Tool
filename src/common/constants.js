export const PATHS = {
  INDEX: '/',
  CREATE_MANIFEST: '/create-manifest',
  EDIT_MANIFEST: '/edit-manifest',
  PROCESSING: '/processing',
  CREATE_GEOREFERENCE: '/create-georeference',
  LAYERS: '/layers',
  LEVELS: '/levels',
  INVALID_PATH: '/*',
};

export const ROUTE_NAME_BY_PATH = {
  [PATHS.INDEX]: 'home',
  [PATHS.CREATE_MANIFEST]: 'create',
  [PATHS.EDIT_MANIFEST]: 'edit',
  [PATHS.PROCESSING]: 'processing',
  [PATHS.CREATE_GEOREFERENCE]: 'create',
  [PATHS.LAYERS]: 'create',
  [PATHS.LEVELS]: 'create',
  [PATHS.INVALID_PATH]: 'redirect',
};

export const CONSTANTS = {
  GEO: {
    US: {
      TEXT: 'geography.unitedstates',
      URL: 'https://us.atlas.microsoft.com',
    },
    // remove before moving to github
    US_TEST: {
      TEXT: 'geography.unitedstates.test',
      URL: 'https://us.t-azmaps.azurelbs.com',
    },
    EU: {
      TEXT: 'geography.europe',
      URL: 'https://eu.atlas.microsoft.com',
    }
  }
};

export const TRUNCATE_FRACTION_DIGITS = 8;