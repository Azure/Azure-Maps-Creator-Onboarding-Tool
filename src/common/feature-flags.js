const flags = {
  onboardingEnabled: false,
};

export default flags;