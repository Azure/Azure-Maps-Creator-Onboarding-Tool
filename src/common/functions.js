/**
 * Given a single path name, returns a list of all path names from root to the given path name.
 * Ex. Given '/create/hello', returns ['/', '/create', '/create/hello']
 * @param {String} pathName the path name of the URL
 * @returns {String[]} an array of path names
 */
export const getSplitPaths = (pathName) => {
  if (typeof pathName !== 'string' || pathName === '') { return []; }
  if (pathName === '/') { return ['/']; }

  const pathNameLowerCase = pathName.toLowerCase();
  const splits = pathNameLowerCase.split('/');
  const splitPaths = [];
  for (let i = splits.length - 1; i > 0; i--) {
    let stringBuilder = '';
    for (let j = 1; j <= i; j++) {
      stringBuilder += '/' + splits[j];
    }
    splitPaths.push(stringBuilder);
  }
  splitPaths.push('/');
  splitPaths.reverse();

  return splitPaths;
};