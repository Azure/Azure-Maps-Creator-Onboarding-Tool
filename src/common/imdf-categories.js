export const imdfCategories = [
  'auditorium',
  'brick',
  'classroom',
  'column',
  'concrete',
  'conferenceroom',
  'drywall',
  'elevator',
  'escalator',
  'fieldofplay',
  'firstaid',
  'fitnessroom',
  'foodservice',
  'footbridge',
  'glass',
  'huddleroom',
  'kitchen',
  'laboratory',
  'library',
  'lobby',
  'lounge',
  'mailroom',
  'mothersroom',
  'movietheater',
  'movingwalkway',
  'nonpublic',
  'office',
  'opentobelow',
  'parking',
  'phoneroom',
  'platform',
  'privatelounge',
  'ramp',
  'recreation',
  'restroom',
  'restroom.family',
  'restroom.female',
  'restroom.female.wheelchair',
  'restroom.male',
  'restroom.male.wheelchair',
  'restroom.transgender',
  'restroom.transgender.wheelchair',
  'restroom.unisex',
  'restroom.unisex.wheelchair',
  'restroom.wheelchair',
  'road',
  'room',
  'serverroom',
  'shower',
  'smokingarea',
  'stairs',
  'steps',
  'storage',
  'structure',
  'terrace',
  'theater',
  'unenclosedarea',
  'unspecified',
  'vegetation',
  'waitingroom',
  'walkway',
  'walkway.island',
  'wood',
];
