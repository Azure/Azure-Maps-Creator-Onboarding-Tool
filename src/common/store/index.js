export { getDomain, useUserStore } from './user.store';
export { useResponseStore } from './response.store';
export { progressBarSteps, progressBarStepsByKey, useCompletedSteps, useProgressBarStore } from './progress-bar-steps';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';