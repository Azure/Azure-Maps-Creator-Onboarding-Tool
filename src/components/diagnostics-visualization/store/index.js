import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

// Middlewares go here
const middleware = {};

const configureStore = (initialState) => createStore(
  rootReducer,
  initialState,
  // Redux Dev Tools Middleware
  composeWithDevTools(applyMiddleware(...middleware)),
);

// pass an optional param to rehydrate state on app start
const store = configureStore();

// export store singleton instance
export default store;
