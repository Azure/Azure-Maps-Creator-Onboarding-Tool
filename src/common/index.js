import { CONSTANTS, PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { CONSTANTS, PATHS, ROUTE_NAME_BY_PATH };