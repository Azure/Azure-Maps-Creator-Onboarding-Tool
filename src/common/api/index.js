// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};