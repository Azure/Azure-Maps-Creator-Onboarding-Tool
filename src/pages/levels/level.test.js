import { fireEvent, render, screen } from '@testing-library/react';

import Level from './level';
import { useLevelsStore, useProgressBarStore } from 'common/store';

const mockLevel = {
  filename: 'kitchen.dwg',
  levelName: 'Kitchen',
  ordinal: '5',
};

describe('Level component', () => {
  beforeEach(() => {
    useProgressBarStore.setState({
      isErrorShown: false,
    });
  });

  it('should render component', () => {
    const view = render(<Level level={mockLevel} />);
    expect(view).toMatchSnapshot();
  });

  it('should show error', () => {
    const emptyLevel = { filename: 'kitchen.dwg', levelName: '', ordinal: '' };
    useProgressBarStore.setState({
      isErrorShown: true,
    });

    const view = render(<Level level={emptyLevel} />);
    expect(view).toMatchSnapshot();
  });

  it('should call setOrdinal on ordinal change', async () => {
    const state = useLevelsStore.getState();
    const spy = jest.spyOn(state, 'setOrdinal');
    render(<Level level={mockLevel} />);
    const input = await screen.findByDisplayValue('5');

    fireEvent.change(input, {
      target: {
        value: '6',
      }
    });
    expect(spy).toHaveBeenCalledWith( 'kitchen.dwg', '6');
  });

  it('should call setLevelName on levelName change', async () => {
    const state = useLevelsStore.getState();
    const spy = jest.spyOn(state, 'setLevelName');
    render(<Level level={mockLevel} />);
    const input = await screen.findByDisplayValue('Kitchen');

    fireEvent.change(input, {
      target: {
        value: 'Ground',
      }
    });
    expect(spy).toHaveBeenCalledWith( 'kitchen.dwg', 'Ground');
  });

  it('should show error message when ordinal is not unique', () => {
    useLevelsStore.setState({
      levels: [mockLevel, { filename: 'filename', levelName: 'levelName', ordinal: '5' }],
    });
    const view = render(<Level level={mockLevel} />);
    expect(view).toMatchSnapshot();
  });

  it('should show error message when level name contains invalid chars', () => {
    const veryLongName = 'asdfjhclsadfhbcalskjdfhcblksdfhcbalksdfhjcbalskdfjhcbalskdfhcblaskfdjhcblaskfjhcbalskfhjcbalskfhjcblaskjfhcbasdlfkhjcb';
    const level = { filename: 'filename', levelName: veryLongName, ordinal: '5' };
    const view = render(<Level level={level} />);
    expect(view).toMatchSnapshot();
  });
});