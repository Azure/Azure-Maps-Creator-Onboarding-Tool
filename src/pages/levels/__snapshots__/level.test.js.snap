// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Level component should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            file.name
          </div>
          <div
            class="ms-TextField is-disabled emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-111"
              >
                <input
                  aria-invalid="false"
                  class="ms-TextField-field field-112"
                  disabled=""
                  id="TextField0"
                  readonly=""
                  type="text"
                  value="kitchen.dwg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            level.name
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-121"
              >
                <input
                  aria-invalid="false"
                  aria-label="level.name.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField5"
                  type="text"
                  value="Kitchen"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            ordinal
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-121"
              >
                <input
                  aria-invalid="false"
                  aria-label="ordinal.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField10"
                  type="text"
                  value="5"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          file.name
        </div>
        <div
          class="ms-TextField is-disabled emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-111"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-112"
                disabled=""
                id="TextField0"
                readonly=""
                type="text"
                value="kitchen.dwg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          level.name
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-121"
            >
              <input
                aria-invalid="false"
                aria-label="level.name.of.file"
                class="ms-TextField-field field-122"
                id="TextField5"
                type="text"
                value="Kitchen"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          ordinal
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-121"
            >
              <input
                aria-invalid="false"
                aria-label="ordinal.of.file"
                class="ms-TextField-field field-122"
                id="TextField10"
                type="text"
                value="5"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Level component should show error message when level name contains invalid chars 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            file.name
          </div>
          <div
            class="ms-TextField is-disabled emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-111"
              >
                <input
                  aria-invalid="false"
                  class="ms-TextField-field field-112"
                  disabled=""
                  id="TextField60"
                  readonly=""
                  type="text"
                  value="filename"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            level.name
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-126"
              >
                <input
                  aria-describedby="TextFieldDescription66"
                  aria-invalid="true"
                  aria-label="level.name.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField65"
                  type="text"
                  value="asdfjhclsadfhbcalskjdfhcblksdfhcbalksdfhjcbalskdfjhcbalskdfhcblaskfdjhcblaskfjhcbalskfhjcbalskfhjcblaskjfhcbasdlfkhjcb"
                />
              </div>
            </div>
            <span
              id="TextFieldDescription66"
            >
              <div
                role="alert"
              />
            </span>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            ordinal
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-126"
              >
                <input
                  aria-describedby="TextFieldDescription71"
                  aria-invalid="true"
                  aria-label="ordinal.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField70"
                  type="text"
                  value="5"
                />
              </div>
            </div>
            <span
              id="TextFieldDescription71"
            >
              <div
                role="alert"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          file.name
        </div>
        <div
          class="ms-TextField is-disabled emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-111"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-112"
                disabled=""
                id="TextField60"
                readonly=""
                type="text"
                value="filename"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          level.name
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-126"
            >
              <input
                aria-describedby="TextFieldDescription66"
                aria-invalid="true"
                aria-label="level.name.of.file"
                class="ms-TextField-field field-122"
                id="TextField65"
                type="text"
                value="asdfjhclsadfhbcalskjdfhcblksdfhcbalksdfhjcbalskdfjhcbalskdfhcblaskfdjhcblaskfjhcbalskfhjcbalskfhjcblaskjfhcbasdlfkhjcb"
              />
            </div>
          </div>
          <span
            id="TextFieldDescription66"
          >
            <div
              role="alert"
            />
          </span>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          ordinal
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-126"
            >
              <input
                aria-describedby="TextFieldDescription71"
                aria-invalid="true"
                aria-label="ordinal.of.file"
                class="ms-TextField-field field-122"
                id="TextField70"
                type="text"
                value="5"
              />
            </div>
          </div>
          <span
            id="TextFieldDescription71"
          >
            <div
              role="alert"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Level component should show error message when ordinal is not unique 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            file.name
          </div>
          <div
            class="ms-TextField is-disabled emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-111"
              >
                <input
                  aria-invalid="false"
                  class="ms-TextField-field field-112"
                  disabled=""
                  id="TextField45"
                  readonly=""
                  type="text"
                  value="kitchen.dwg"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            level.name
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-121"
              >
                <input
                  aria-invalid="false"
                  aria-label="level.name.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField50"
                  type="text"
                  value="Kitchen"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="emotion-1"
        >
          <div
            class="emotion-2"
          >
            ordinal
          </div>
          <div
            class="ms-TextField emotion-3 root-110"
          >
            <div
              class="ms-TextField-wrapper"
            >
              <div
                class="ms-TextField-fieldGroup fieldGroup-126"
              >
                <input
                  aria-describedby="TextFieldDescription56"
                  aria-invalid="true"
                  aria-label="ordinal.of.file"
                  class="ms-TextField-field field-122"
                  id="TextField55"
                  type="text"
                  value="5"
                />
              </div>
            </div>
            <span
              id="TextFieldDescription56"
            >
              <div
                role="alert"
              />
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  max-width: 45rem;
  border-bottom: 1px solid #ccc;
  padding-top: 1.25rem;
}

.emotion-0:first-child {
  padding-top: 0;
}

.emotion-0:last-child {
  border-bottom: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1.25rem;
}

.emotion-2 {
  height: 1.5rem;
  line-height: 1.5rem;
  min-width: 7rem;
  max-width: 15rem;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.813rem;
  width: 15.625rem;
}

.emotion-2:after {
  content: " *";
  color: #A4262C;
  white-space: pre;
}

.emotion-3 {
  width: 100%;
  max-width: 30rem;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          file.name
        </div>
        <div
          class="ms-TextField is-disabled emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-111"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-112"
                disabled=""
                id="TextField45"
                readonly=""
                type="text"
                value="kitchen.dwg"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          level.name
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-121"
            >
              <input
                aria-invalid="false"
                aria-label="level.name.of.file"
                class="ms-TextField-field field-122"
                id="TextField50"
                type="text"
                value="Kitchen"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="emotion-1"
      >
        <div
          class="emotion-2"
        >
          ordinal
        </div>
        <div
          class="ms-TextField emotion-3 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-126"
            >
              <input
                aria-describedby="TextFieldDescription56"
                aria-invalid="true"
                aria-label="ordinal.of.file"
                class="ms-TextField-field field-122"
                id="TextField55"
                type="text"
                value="5"
              />
            </div>
          </div>
          <span
            id="TextFieldDescription56"
          >
            <div
              role="alert"
            />
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
